package sup2;

import java.util.Scanner;

public class ExceptionHandle {
	// 필드로 Sanner를 준비하면 좀 더 안정적으로 관리할 수 있다.
	public static Scanner sc = new Scanner(System.in);
	public static void main(String[] args) {
		// Exception : 예외.
		// 이미 사전에 프로그램이 강종되는 것을 예방하기 위해 준비한 보고체계
		// 구조는 크게 3가지로 이루어짐
		// 1) try : 감시하는 구역을 설정함
		// 2) catch : 구역에 내에서 발생한 예외를 접수하고
		//            그에 대응되는 코드를 실행함
		// 3) finally : 구역내에서 벌렸던 일을 뒷정리하기 위한 코드구간
		//              입출력관련해서 주로 설정하여 사용함
		//              무조건 실행되는 코드다.
		
		// 사용방식1. 에러원인을 파악해 해결하고, 예외처리를 제거함
		// 사용방식2. 
		int num1 = sc.nextInt();
		int num2 = sc.nextInt();
		try {
			int result = num1 / num2;
			System.out.println("결과 : " + result);
		}
		catch(ArithmeticException e) {
			// 예외 객체를 접수하고, 이를 출력할 수 있으며
			// 이 때 에러의 발생원인 등을 확인할 수 있다.
			System.out.println(e);
		}
		finally {
			System.out.println("안전하게 종료할께요!");
			sc.close();
		}
	}
}




